sprocs/variants_insert.sql

```diff
@@ -83,19 +83,30 @@ BEGIN
         real_user_id := user_id;
     END IF;
 
-    -- check consistency of question_id and course_instance_id
-    IF real_question_id IS NOT NULL AND real_course_instance_id IS NOT NULL THEN
-        SELECT c.id
-        INTO course_id
+    -- check consistency of question_id and course_id
+    SELECT q.id
+    INTO real_question_id
+    FROM
+        questions AS q
+    WHERE
+        q.id = real_question_id
```

Comment:
I am ever-so-slightly weirded out by the same variable being both an input to and output from a query, I did a double take when I saw this. Perhaps it would be slightly clearer to do something like this?

```sql
SELECT FROM questions ...;
IF NOT FOUND THEN RAISE ...;
```

cc @mwest1066 I'm not sure if this is a smell I should be worried about.

Reply:
I agree that looks better, I'll change it.

I'm pretty inexperienced with PL/pgSQL so I didn't think to use that, even though I think I may have seen it before.

Reply:
Yeah, I saw that and was also slightly unhappy but I didn't flag it. I'd certainly prefer to avoid this double-use of the same variable.

Reply:
huh, what you are suggesting is actually not allowed, I get a SQL error:

> If you want to discard the results of a SELECT, use PERFORM instead

Reply:
@mwest1066 what do recommend the best thing to do here stylistically, given the error I got?

Reply:
This is a small quirk of plpgsql -- if you don't want to directly save the results of the SELECT then you need to write it as PERFORM instead. We already do this elsewhere. For example, here we do a perform: https://github.com/PrairieLearn/PrairieLearn/blob/236d6813961354faa213e15e050197ba2564b52a/sprocs/ip_to_mode.sql#L25 just so we can test for FOUND: https://github.com/PrairieLearn/PrairieLearn/blob/236d6813961354faa213e15e050197ba2564b52a/sprocs/ip_to_mode.sql#L37

Reply:
Thanks for the code pointer! I couldn't figure out the syntax. 

All fixed now.

Reply:
lgtm